cmd/version/version.go

```diff
@@ -9,7 +9,7 @@ import (
 var (
 	// TODO: This goes away when control-plane-operator becomes another component
 	// in the OCP payload.
-	HyperShiftImage = "registry.ci.openshift.org/hypershift/hypershift:latest"
+	HyperShiftImage = "quay.io/hypershift/hypershift:latest"
```

Comment:
This image is also referenced [here](https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L1792) - do we need to update that, or remove that hard-coded reference to refer to this variable?

I'm also interested to know what the plan is for the CAPI/CAPA/CAPK/network-proxy images which are also currently consumed from `registry.ci.openshift.org` ?  

(context - I started looking at how to deploy hypershift in a disconnected/ipv6 env and those hard-coded image references make it difficult since you can't override them like the hypershift image)

Reply:
@hardys thanks for taking a look!

>This image is also referenced here - do we need to update that, or remove that hard-coded reference to refer to this variable?

The logic in https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L1792 is somewhat confusing at a glance; the `registry.ci.openshift.org/hypershift/hypershift` value is only used with 4.8 versions. The 4.8 builds are remaining as-is and will continue to be published to that location.

I went ahead and updated that code to move the pull spec value closer to where it's actually used. Hopefully that makes it more clear.

>I'm also interested to know what the plan is for the CAPI/CAPA/CAPK/network-proxy images which are also currently consumed from registry.ci.openshift.org ?

@enxebre, do you know what our plans are for these? Seems like we could move all our public images to quay if we wanted to.

>(context - I started looking at how to deploy hypershift in a disconnected/ipv6 env and those hard-coded image references make it difficult since you can't override them like the hypershift image)

Every image you mentioned can be user defined one way or another — is there a specific one you're having trouble with? I can point you to the right place.

Reply:
> The logic in https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L1792 is somewhat confusing at a glance; the `registry.ci.openshift.org/hypershift/hypershift` value is only used with 4.8 versions. The 4.8 builds are remaining as-is and will continue to be published to that location.

Ah I see, thanks for the clarification! 

> Every image you mentioned can be user defined one way or another — is there a specific one you're having trouble with? I can point you to the right place.

I probably need to take another look - before the break I was testing like

```
hypershift create cluster none ... --control-plane-operator-image virthost.ostest.test.metalkube.org:5000/localimages/hypershift:latest --release-image virthost.ostest.test.metalkube.org:5000/localimages/local-release-image:latest
```

I couldn't see a way to override those additional images, but I probably need to revisit - any pointers much appreciated! :)

Obviously this all gets way simpler if all we have to specify is the release-image mirrored to a local registry, hence my questions about getting those into the payload (I guess the same images from the single-cluster CAPI work can be reused if/when available?)

Reply:
I just took a quick scan through the code to try and map out the images which are configurable on a per hostedcluster basis. This is not yet exhaustive, but for example, the following annotations on HostedCluster are honored and can be used to override images:

```
hypershift.openshift.io/control-plane-operator-image
hypershift.openshift.io/capi-manager-image
hypershift.openshift.io/aws-kms-provider-image
hypershift.openshift.io/ibmcloud-kms-provider-image
hypershift.openshift.io/portieris-image
hypershift.openshift.io/machine-approver-image
hypershift.openshift.io/cluster-autoscaler-image
hypershift.openshift.io/konnectivity-agent-image
hypershift.openshift.io/konnectivity-server-image
```

These are all unsupported/undocumented at the moment. Notably absent is a way to override the CAPI AWS provider image, currently hard-coded to `registry.ci.openshift.org/hypershift/cluster-api-aws-controller:v1.1.0`.

There are also some flags on control-plane-operator which allow overriding some images that normally come from the payload, but you can't use those through hostedcluster annotations.

Based on this convo and with the increasing number of contributors and development use cases, I'm thinking we probably need to come up with a more generic and documented way to do this stuff that covers all the possible images. I'll put together an issue to link here so we can take the conversation elsewhere.

Thanks!